R/AllGenerics.R

```diff
@@ -4,6 +4,15 @@ setGeneric("annotateMz", function(object, compounds, ...)
 setGeneric("compoundVariables", function(object, ...)
     standardGeneric("compoundVariables"))
 
+setGeneric("ionVariables", function(object, ...)
+  standardGeneric("ionVariables"))
+
+setGeneric("insertIon", function(object, ions, ...)
+  standardGeneric("insertIon"))
+
+setGeneric("ions", function(object, ...)
```

Comment:
Can you please remove this generic and import the `ions` method from `ProtGenerics` (i.e. in the code where you have `setMethod("ions", "IonDb"...` have the `#' @importMethodsFrom ProtGenerics ions`).

Reply:
I have removed the generic and added `#' @importMethodsFrom ProtGenerics ions` above the `setMethod("ions", "IonDb"... ` but now when I try to document with `devtools`or `roxygen` I get the following error:

Errore in setMethod("ions", "IonDb", function(object, columns = ionVariables(object),  : 
  no existing definition for function ‘ions’
 
Do you have any idea what the reason could be?

Reply:
this also happens sometimes to me - no idea why, I guess `devtools` caching some old information or so. try deleting the NAMESPACE file and then run `devtools::document()` again

Reply:
Thanks, that's stange, I tried to do that but then I obtained a similar error with `Spectra`instead of `ions`and the NAMESPACE was not created at all. So, I restored it and added maually a line with `importMethodsFrom(ProtGenerics,ions)` although the namespace starts with "don't edit by hand" 😅. Hope that is ok. Now it works

Reply:
OMG, yes, I am also experiencing more and more of these problems (maybe there was an update to `devtools`?) - actually, adding the import manually was the next thing I was to suggest :)